packages/codebytes/src/index.tsx

```diff
@@ -35,13 +35,23 @@ const EditorContainer = styled(Background)<EditorStyleProps>(
 export const CodeByteEditor: React.FC<CodeByteEditorProps> = ({
   text: initialText,
   language: initialLanguage,
-  hideCopyButton,
+  hideCopyButton = false,
   isIFrame = false,
   on = {},
   snippetsBaseUrl = process.env.CONTAINER_API_BASE,
 }) => {
-  const [text, setText] = useState<string>(initialText);
-  const [language, setLanguage] = useState<languageOption>(initialLanguage);
+  const [text, setText] = useState<string>(initialText ?? '');
+  const [language, setLanguage] = useState<languageOption>(
+    initialLanguage ?? ''
+  );
+
+  /* Resetting the state when the props change. If the language changes and the text is not provided output helloworld  */
+  useEffect(() => {
+    if (language) {
+      setText(initialText ?? (helloWorld[language] || ''));
+    }
+  }, [language, initialText]);
```

Comment:
This ensures the state resets when the props change, i.e. if the language changes and the text is not provided, it will output helloworld.